inst/include/mode/r/mode.hpp

```diff
@@ -279,5 +279,36 @@ void mode_reorder(SEXP ptr, cpp11::sexp r_index) {
   obj->reorder(index);
 }
 
+template <typename T>
+SEXP mode_resample(SEXP ptr, cpp11::doubles r_weights) {
+  cpp11::stop("Can't use resample with mode models");
+  return R_NilValue; // unreachable
+}
+
+// This is the approach used in dust, but though it switches using the
+// value of the data type, which we don't support yet.
+inline void disable_method(const char * name) {
+  cpp11::stop("The '%s' method is not supported for this class", name);
+}
+
+template <typename T>
+void mode_set_data(SEXP ptr, cpp11::list r_data, bool data_is_shared) {
```

Comment:
why is this one void and the others return `cpp11:sexp`?

Reply:
It returns an R `NULL`, cpp11 will do the conversions for us to get there. The other methods that return `NULL` should probably do that too (it's a small thing though and we can iron over that when we merge the two packages)

Reply:
oh, within this PR sorry - that's because the set data one would never return, but the others do return for models that support interacting with data - we just never return in mode